src/librustc/ty/query/caches.rs

```diff
@@ -54,13 +58,18 @@ pub(crate) trait QueryCache<K, V>: Default {
 pub struct DefaultCacheSelector;
 
 impl<K: Eq + Hash, V: Clone> CacheSelector<K, V> for DefaultCacheSelector {
-    type Cache = DefaultCache;
+    type Cache = DefaultCache<()>;
 }
 
-#[derive(Default)]
-pub struct DefaultCache;
+pub struct DefaultCache<D>(PhantomData<D>);
```

Comment:
What does this generic parameter `D` do?

Reply:
Avoids compiler errors for the `pub type LocalDenseDefIdCacheSelector<V>` alias.